blaze/compute/tests/test_sparksql.py

```diff
@@ -0,0 +1,248 @@
+    expr = db.t[db.t.amount > 50]
+    result = compute(expr, ctx)
+    expected = compute(expr, {db: {'t': df}})
+    assert list(map(set, into(list, result))) == list(map(set, into(list,
+                                                                    expected)))
+
+
+def test_selection_field(ctx, db):
+    expr = db.t[db.t.amount > 50].name
+    result = compute(expr, ctx)
+    expected = compute(expr, {db: {'t': df}})
+    assert into(set, result, dshape=expr.dshape) == into(set, expected)
+
+
+@pytest.mark.parametrize(['field', 'reduction'],
+                         itertools.product(['id', 'amount'], ['sum', 'max']))
+def test_reductions(ctx, db, field, reduction):
+    expr = getattr(getattr(db.t, field), reduction)()
+    result = compute(expr, ctx)
+    expected = compute(expr, {db: {'t': df}})
+    assert into(list, result)[0][0] == expected
+
+
+def test_column_arithmetic(ctx, db):
+    expr = db.t.amount + 1
+    result = compute(expr, ctx)
+    expected = compute(expr, {db: {'t': df}})
+    assert into(set, result, dshape=expr.dshape) == into(set, expected)
+
+
+@pytest.mark.parametrize('function', [pytest.mark.xfail(sin),
+                                      pytest.mark.xfail(exp)])
+def test_math(ctx, db, function):
+    expr = function(db.t.amount)
+    result = compute(expr, ctx)
+    expected = compute(expr, {db: {'t': df}})
+    assert into(set, result, dshape=expr.dshape) == into(set, expected)
+
+
+@pytest.mark.parametrize(['field', 'ascending'],
+                         itertools.product(['name', 'id'], [True, False]))
+def test_sort(ctx, db, field, ascending):
+    expr = db.t.sort(field, ascending=ascending)
+    result = compute(expr, ctx)
+    expected = compute(expr, {db: {'t': df}})
+    assert list(map(set, into(list, result))) == list(map(set, into(list,
+                                                                    expected)))
+
+
+@pytest.mark.xfail
+def test_map(ctx, db):
+    expr = db.t.id.map(lambda x: x + 1, 'int')
+    result = compute(expr, ctx)
+    expected = compute(expr, {db: {'t': df}})
+    assert into(set, result, dshape=expr.dshape) == into(set, expected)
+
+
+@pytest.mark.parametrize(['grouper', 'reducer', 'reduction'],
+                         itertools.product(['name', 'id'],
+                                           ['id', 'amount'],
+                                           ['sum', 'count']))
+def test_by(ctx, db, grouper, reducer, reduction):
+    t = db.t
+    expr = by(getattr(t, grouper), total=getattr(getattr(t, reducer),
+                                                 reduction)())
+    result = compute(expr, ctx)
+    expected = compute(expr, {db: {'t': df}})
+    assert (set(map(frozenset, into(list, result))) ==
+            set(map(frozenset, into(list, expected))))
+
+
+@pytest.mark.parametrize(['reducer', 'reduction'],
+                         itertools.product(['id', 'name'],
+                                           ['count', 'nunique']))
```

Comment:
The use of parameters seems appropriate to me here.  Might even want to go further with expression up above if it can be done simply.